packages/flutter_tools/lib/src/compile.dart

```diff
@@ -300,6 +311,27 @@ class KernelCompiler {
     }
     return null;
   }
+
+  void generateMainDartWithPluginRegistrant(String currentMainUri, File newMainDart) {
+    try {
+    newMainDart.writeAsStringSync('''
+import '$currentMainUri' as entrypoint;
+
+// The plugin registrant is called by dart:ui.
+@pragma('vm:entry-point')
+void _registerPlugins() {
+  // TODO(egarciad): Call each plugin's main class registerWith();
+  print('_registerPlugins is called');
+}
+
+void main() {
```

Comment:
If you're going to generate a synthetic entrypoint, then I don't think you need the pragma or anything else special - just invoke _registerPlugins() before calling entrypoint.main.

Reply:
main can be overridden by the embedding at runtime.

Reply:
right :) . lots of moving parts here.